src/test/java/org/apache/datasketches/quantilescommon/IncludeMinMaxTest.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.datasketches.quantilescommon;
+
+import static org.apache.datasketches.common.Util.LS;
+import static org.apache.datasketches.quantilescommon.IncludeMinMax.DoublesPair;
+import static org.apache.datasketches.quantilescommon.IncludeMinMax.FloatsPair;
+import static org.apache.datasketches.quantilescommon.IncludeMinMax.ItemsPair;
+import static org.testng.Assert.assertEquals;
+
+import java.util.Comparator;
+
+import org.testng.annotations.Test;
+
+/**
+ * Checks the IncludeMinMax process
+ */
+public final class IncludeMinMaxTest {
+
+  @Test
+  public static void checkDoublesEndsAdjustment() {
+    final double[] quantiles = {2, 4, 6, 7};
+    final long[] cumWeights = {2, 4, 6, 8};
+
+    final double maxItem = 8;
+    final double minItem = 1;
+    final DoublesPair dPair = IncludeMinMax.includeDoublesMinMax(quantiles, cumWeights, maxItem, minItem);
+    final double[] adjQuantiles = dPair.quantiles;
+    final long[] adjCumWeights = dPair.cumWeights;
+    int len = adjCumWeights.length;
+    printf("%10s %10s" + LS, "Quantiles", "CumWeights");
+    for (int i = 0; i < len; i++) {
+      printf("%10.1f %10d" + LS, adjQuantiles[i], adjCumWeights[i]);
+    }
+    final int topIn = quantiles.length - 1;
+    final int topAdj = adjQuantiles.length - 1;
+    assertEquals(adjQuantiles[topAdj], maxItem);
```

Comment:
i think it'd be good to assert that the adjQuantiles array is 2 longer than the original quantiles array in here, too.

The printfs might be convient when debugging but they're basically no-ops in tests (and I treat them as such when reviewing). We want to make sure we assert on the expected 
end-user behavior. And in this case, that's that the arrays are longer because we add in min and max.

Reply:
Done.